source/templates/secrets.slim

```diff
@@ -6,17 +6,32 @@
 		article
 			ul.env-vars[order-by-drag models="envVarsOfSource.envVars"]
 				li[ng-repeat="envVar in envVarsOfSource.envVars"]
-					form[name="envVarForm"]
+					p.progress-indicator[progress-model="envVar.customProperties.progress"]
+					form[name="envVarForm" ng-if="!envVar.customProperties.progress.isInProgress" autocomplete="off" autocapitalize="none"]
 						.key-value-delete
-							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" skip-drag-order]
-							textarea.value[name="value" msd-elastic ng-model="envVar.value" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
-							button.delete-env-var.rebo.danger[type="button" ng-click="envVarsCtrl.removeEnvVarFromEnvVars(envVar, envVarsOfSource.envVars)"] = svg("cross")
+							input.key[type="text" name="key" ng-model="envVar.key" ng-model-options="{getterSetter: true}" ng-disabled="!envVar.isKeyChangeable" custom-validators="{'key': Variable.isValidKey(envVar.key(), envVarsOfSource.envVars)}" placeholder="#{ data[:strings][:env_vars][:key_placeholder] }" autocomplete="off" autocorrect="off" autocapitalize="none" spellcheck="false" skip-drag-order]
+							.value-with-actions.secret
+								textarea.value[name="value" msd-elastic ng-model="envVarsCtrl.valueGetterSetterForEnvVar(envVar)" ng-class="{'protected' : envVar.isProtected()}" ng-disabled="!envVar.shouldShowValue" ng-model-options="{getterSetter: true}" ng-trim="false" placeholder="#{ data[:strings][:env_vars][:value_placeholder] }" skip-drag-order]
```

Comment:
You didn't test this in offline mode:
- `valueGetterSetterForEnvVar` is not working properly
- the `protected` class should never be applied in offline mode
- `shouldShowValue` should be true in offline mode